### ISSUE TITLE
Issue in compiling with xcode 8 - for legacy support - swift 2.3

### ISSUE THREAD

**Issue by harit** [onevcat/APNGKit#22]
Hi!

I am trying to use the library. I get error for below:
`/APNGKit/Frame.swift:75:73: Nil is not compatible with expected argument type 'CGDataProviderReleaseDataCallback' (aka '@convention(c) (UnsafeMutablePointer<()>, UnsafePointer<()>, Int) -> ()')`

Tried this for most versions of library - 0.1.0 to 0.1.4.

Please suggest way to deal with this. My project is in Swift 2.3, so need library compiled with 2.3 to be used.

Thanks!

**Comment by lumenlunae** [comment]
Have you tried using version 0.2.2? It's in the readme that it supports 2.2 (not sure on 2.3)

**Comment by harit** [comment]
yes, used the exact version via pods and also set Use Legacy Swift Language Version flag to YES.

**Comment by onevcat** [comment]
:( Currently no Swift 2.3 support. I will try to add it later today.

**Comment by onevcat** [comment]
I pushed the change into the swift2.3 branch. Now you can use this to integrate APNGKit to your Swift 2.3 project.

```
pod 'APNGKit', :git => 'https://github.com/onevcat/APNGKit.git', :branch => 'swift2.3'
```

**Comment by d1248966459** [comment]
喵神,我需要在swift2.3开发的私有库中依赖APNGKit，但像这样没有release的版本似乎是没办法用
>  s.dependency 'APNGKIt' :git => 'https://github.com/onevcat/APNGKit.git', :branch => 'swift2.3'
这种方式引入的，可否考虑下提交一个swift2.3的版本？麻烦啦

**Comment by onevcat** [comment]
I added a tag `0.2.3` for it.

**Comment by d1248966459** [comment]
thanks

**Comment by nlucky** [comment]
喵神请教下，我试了下
> s.dependency 'APNGKIt' :git => 'https://github.com/onevcat/APNGKit.git', :tag  => '0.2.3'

发现还是不行的。我看cocoapods里有类似的issue，但并没有给出一个比较好的解决方法，推荐的还是用一个release的版本来添加依赖关系。请问我该如何利用tag，来实现在私有库中添加依赖关系呢？

**Comment by d1248966459** [comment]
喵神请教下，我试了下
> s.dependency 'APNGKIt' :git => 'https://github.com/onevcat/APNGKit.git', :tag  => '0.2.3'

发现还是不行的。我看cocoapods里有类似的issue，但并没有给出一个比较好的解决方法，推荐的还是用一个release的版本来添加依赖关系。请问我该如何利用tag，来实现在私有库中添加依赖关系呢？

**Comment by onevcat** [comment]
@d1248966459 I also published the 0.2.3 to CocoaPods, now you should be able to use it as a dependency. :) 

https://github.com/CocoaPods/Specs/tree/master/Specs/0/6/4/APNGKit/0.2.3

**Comment by d1248966459** [comment]
非常感谢
